contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,902 @@
+
+    motion.votes[_vote] = add(motion.votes[_vote], userRep);
+    motion.repRevealed = add(motion.repRevealed, userRep);
+
+
+    uint256 fractionUserReputation = wdiv(userRep, motion.skillRep);
+    uint256 totalStake = add(motion.stakes[YAY], motion.stakes[NAY]);
+    uint256 voterReward = wmul(wmul(fractionUserReputation, totalStake), voterRewardFraction);
+
+    motion.paidVoterComp = add(motion.paidVoterComp, voterReward);
+    tokenLocking.transfer(token, voterReward, msg.sender, true);
+
+    emit MotionVoteRevealed(_motionId, msg.sender, _vote);
+
+    if (motion.repRevealed == motion.repSubmitted) {
+      motion.events[REVEAL_END] = uint64(now);
+
+      emit MotionEventSet(_motionId, REVEAL_END);
+    }
+  }
+
+  /// @notice Escalate a motion to a higher domain
+  /// @param _motionId The id of the motion
+  /// @param _newDomainId The desired domain of escalation
+  /// @param _childSkillIndex For the current domain, relative to the escalated domain
+  /// @param _key Reputation tree key for the new domain
+  /// @param _value Reputation tree value for the new domain
+  /// @param _branchMask The branchmask of the proof
+  /// @param _siblings The siblings of the proof
+  function escalateMotion(
+    uint256 _motionId,
+    uint256 _newDomainId,
+    uint256 _childSkillIndex,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Motion storage motion = motions[_motionId];
+    require(getMotionState(_motionId) == MotionState.Closed, "voting-rep-motion-not-closed");
+
+    uint256 newDomainSkillId = colony.getDomain(_newDomainId).skillId;
+    uint256 childSkillId = colonyNetwork.getChildSkillId(newDomainSkillId, _childSkillIndex);
+    require(childSkillId == motion.skillId, "voting-rep-invalid-domain-proof");
+
+    motion.events[STAKE_END] = uint64(now + stakePeriod);
+    motion.events[SUBMIT_END] = uint64(now + stakePeriod + submitPeriod);
+    motion.events[REVEAL_END] = uint64(now + stakePeriod + submitPeriod + revealPeriod);
+
+    motion.domainId = _newDomainId;
+    motion.skillId = newDomainSkillId;
+    motion.skillRep = getReputationFromProof(_motionId, address(0x0), _key, _value, _branchMask, _siblings);
+
+    uint256 loser = (motion.votes[NAY] < motion.votes[YAY]) ? NAY : YAY;
+    motion.stakes[loser] = sub(motion.stakes[loser], motion.paidVoterComp);
+    motion.pastVoterComp[loser] = add(motion.pastVoterComp[loser], motion.paidVoterComp);
+    delete motion.paidVoterComp;
+
+    emit MotionEscalated(_motionId, msg.sender, _newDomainId);
+  }
+
+  function finalizeMotion(uint256 _motionId) public {
+    Motion storage motion = motions[_motionId];
+    require(getMotionState(_motionId) == MotionState.Finalizable, "voting-rep-motion-not-executable");
+
+    motion.finalized = true;
+
+    bool canExecute = (
+      motion.stakes[NAY] < motion.stakes[YAY] ||
+      motion.votes[NAY] < motion.votes[YAY]
+    );
```

Comment:
Not quite, since this can only be reached if the YAY side is fully staked OR if there was a past vote. I've added an `assert` to this effect. But basically I'm reading this as: "We execute IF YAY is fully staked and NAY is not, OR if YAY has strictly more votes than NAY."

The change was made _in order to_ make it so that NAYs won in a tie -- in the earlier `<=` implementation, a tie would go to the YAY side, which I realized was incorrect.

Reply:
Ah, right, I follow now - I hadn't clocked that if `YAY` isn't fully staked that prevents the `Finalizable` state being reached.

Reply:
Yup! It's a kind of obscure relationship between `finalizeMotion` and `getMotionState` which is just _perfect_ for getting broken down the road so I've added an assert to be defensive.